rpc_util.go

```diff
@@ -250,10 +343,22 @@ func PerRPCCredentials(creds credentials.PerRPCCredentials) CallOption {
 //
```

Comment:
> If WithCompressor is also set, UseCompressor has higher priority.

This raises an interesting point... we have some [DialOptions](https://godoc.org/google.golang.org/grpc#DialOption) that also can alter per-RPC behavior.  _Especially_ `WithDefaultCallOptions`.

To do this right, we should make sure there are CallOption equivalents for every appropriate DialOption, express them in terms of default CallOptions instead, and then pass those defaults along with the per-call CallOptions to the interceptors.

Reply:
The other dial options (`WithMaxMsgSize` and `WithCodec`) use `WithDefaultCallOptions` under the hood. So exposing these all to the interceptor turned out to be pretty simple.